stellargraph/core/validation.py

```diff
@@ -40,3 +41,22 @@ def require_dataframe_has_columns(name, df, columns):
         raise ValueError(
             f"{name}: expected {comma_sep(columns)} columns, found: {comma_sep(df.columns)}"
         )
+
+
+def require_integer_in_range(x, variable_name, min_val=-np.inf, max_val=np.inf):
+    """
+    A function to verify that a variable is an integer in a specified closed range.
+    Args:
+        x: the variable to check
+        variable_name (str): the name of the variable to print out in error messages
+        min_val: the minimum value that `x` can attain
+        min_val: the maximum value that `x` can attain
+    """
+
+    if not isinstance(x, int):
+        raise TypeError(f"{variable_name}: expected int, found {type(x).__name__}")
+
+    if x < min_val or x > max_val:
+        raise ValueError(
+            f"{variable_name}: expected integer in the range [{min_val}, {max_val}], found {x}"
```

Comment:
With the default bound, this ends up formatted as `[1, inf]`, which is a little hard to read; maybe this could customise the message?

```python
if min_val == -np.inf:
    region = f"less or equal to {max_val}"
elif max_val == np.inf:
    region = f"greater or equal to {min_val}"
else:
    region = f"in the range [{min_val}, {max_val}]"

raise ValueError(f"{variable_name}: expected integer {region}, found {x}"
```

or even `<=` and `>=` instead of the words.